lightning-transaction-sync/src/tests.rs

```diff
@@ -0,0 +1,242 @@
+	let _block_hashes = get_bitcoind().client.generate_to_address(num as u64, &address).unwrap();
+}
+
+fn wait_for_block(min_height: usize) {
+	let mut header = get_electrsd().client.block_headers_subscribe().unwrap();
+	loop {
+		if header.height >= min_height {
+			break;
+		}
+		header = exponential_backoff_poll(|| {
+			get_electrsd().trigger().unwrap();
+			get_electrsd().client.ping().unwrap();
+			get_electrsd().client.block_headers_pop().unwrap()
+		});
+	}
+}
+
+fn exponential_backoff_poll<T, F>(mut poll: F) -> T
+where
+	F: FnMut() -> Option<T>,
+{
+	let mut delay = Duration::from_millis(64);
+	loop {
+		match poll() {
+			Some(data) => break data,
+			None if delay.as_millis() < 512 => delay = delay.mul_f32(2.0),
+			None => {}
+		}
+
+		std::thread::sleep(delay);
+	}
+}
+
+fn premine() {
+	PREMINE.get_or_init(|| {
+		let _miner = MINER_LOCK.lock().unwrap();
+		generate_blocks_and_wait(101);
+	});
+}
+
+#[derive(Debug)]
+enum TestConfirmableEvent {
+	Confirmed(Txid, BlockHash, u32),
+	Unconfirmed(Txid),
+	BestBlockUpdated(BlockHash, u32),
+}
+
+struct TestConfirmable {
+	pub confirmed_txs: Mutex<HashMap<Txid, (BlockHash, u32)>>,
+	pub unconfirmed_txs: Mutex<HashSet<Txid>>,
+	pub best_block: Mutex<(BlockHash, u32)>,
+	pub events: Mutex<Vec<TestConfirmableEvent>>,
+}
+
+impl TestConfirmable {
+	pub fn new() -> Self {
+		let genesis_hash = genesis_block(Network::Regtest).block_hash();
+		Self {
+			confirmed_txs: Mutex::new(HashMap::new()),
+			unconfirmed_txs: Mutex::new(HashSet::new()),
+			best_block: Mutex::new((genesis_hash, 0)),
+			events: Mutex::new(Vec::new()),
+		}
+	}
+}
+
+impl Confirm for TestConfirmable {
+	fn transactions_confirmed(&self, header: &BlockHeader, txdata: &TransactionData<'_>, height: u32) {
+		for (_, tx) in txdata {
+			let txid = tx.txid();
+			let block_hash = header.block_hash();
+			self.confirmed_txs.lock().unwrap().insert(txid, (block_hash, height));
+			self.unconfirmed_txs.lock().unwrap().remove(&txid);
+			self.events.lock().unwrap().push(TestConfirmableEvent::Confirmed(txid, block_hash, height));
+		}
+	}
+
+	fn transaction_unconfirmed(&self, txid: &Txid) {
+		self.unconfirmed_txs.lock().unwrap().insert(*txid);
+		self.confirmed_txs.lock().unwrap().remove(txid);
+		self.events.lock().unwrap().push(TestConfirmableEvent::Unconfirmed(*txid));
+	}
+
+	fn best_block_updated(&self, header: &BlockHeader, height: u32) {
+		let block_hash = header.block_hash();
+		*self.best_block.lock().unwrap() = (block_hash, height);
+		self.events.lock().unwrap().push(TestConfirmableEvent::BestBlockUpdated(block_hash, height));
+	}
+
+	fn get_relevant_txids(&self) -> Vec<(Txid, Option<BlockHash>)> {
+		self.confirmed_txs.lock().unwrap().iter().map(|(&txid, (hash, _))| (txid, Some(*hash))).collect::<Vec<_>>()
+	}
+}
+
+pub struct TestLogger {}
+
+impl Logger for TestLogger {
+	fn log(&self, _record: &Record) {}
+}
+
+#[test]
+fn test_esplora_syncs() {
```

Comment:
Should we use a `tests` directory? As per: https://doc.rust-lang.org/book/ch11-03-test-organization.html#integration-tests